.github/workflows/test_pr_changes.yml

```diff
@@ -12,6 +12,129 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  detect-changes-new:
+    if: github.event.pull_request.draft == false
+    runs-on: ghcr.io/cirruslabs/macos-runner:sonoma
+    env:
+      GITHUB_ACCESS_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      GITHUB_USERNAME: ${{ secrets.ZENKINS_USERNAME }}
+      SLACK_WEBHOOK_URL: ${{ secrets.WIRE_IOS_CI_WEBHOOK }}
+
+    steps:
+      - uses: actions/checkout@v3
+      
+      - name: Retrieve Xcode version
+        run: |
+          echo "XCODE_VERSION=$(cat .xcode-version)" >> $GITHUB_OUTPUT
+        id: xcode-version
+
+      - uses: maxim-lobanov/setup-xcode@v1.5.1
+        with:
+          xcode-version: ${{ steps.xcode-version.outputs.XCODE_VERSION }}
+
+      - name: Load .env file
+        uses: xom9ikk/dotenv@v2
+        with:
+            path: fastlane/
+
+      - name: Restore Carthage Cache
+        uses: actions/cache@v3
+        id: cache-carthage
+        with:
+          path: Carthage
+          key: ${{ runner.os }}-xcode${{ steps.xcode-version.outputs.XCODE_VERSION }}-carthage-${{ hashFiles('Cartfile.resolved') }}
+
+      - name: Bootstrap Carthage if no cache
+        if: steps.cache-carthage.outputs.cache-hit != 'true'
+        run: ./scripts/carthage.sh bootstrap --platform ios --use-xcframeworks
+
+      - name: Restore Bundler Cache
+        uses: actions/cache@v3
+        id: cache-bundler
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-bundler-${{ hashFiles('Gemfile.lock') }}
+
+      - name: Setup workspace
+        run: |
+          ./setup.sh
+      
+      - name: Setup simulator
+        run: |
+          bundle exec fastlane prepare_for_tests
+      
+      - name: Setup selective testing
+        run: |
+          mint run mikeger/XcodeSelectiveTesting@0.9.3 --base-branch $GITHUB_BASE_REF
```

Comment:
Running this would update the AllTests test plan, disabling unnecessary test targets.
In a final integration (if you find this useful) it would be necessary to cache the mint build folder to save the tool's compilation time.